tests/models/llava/test_processor_llava.py

```diff
@@ -11,18 +11,57 @@
 # WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 # See the License for the specific language governing permissions and
 # limitations under the License.
+import json
+import tempfile
 import unittest
 
+from transformers import AutoTokenizer, LlamaTokenizerFast, LlavaProcessor
 from transformers.testing_utils import require_vision
 from transformers.utils import is_vision_available
 
+from ...test_processing_common import ProcessorTesterMixin
+
 
 if is_vision_available():
-    from transformers import AutoTokenizer, LlavaProcessor
+    from transformers import CLIPImageProcessor
 
 
 @require_vision
-class LlavaProcessorTest(unittest.TestCase):
+class LlavaProcessorTest(ProcessorTesterMixin, unittest.TestCase):
+    processor_class = LlavaProcessor
+
+    def setUp(self):
+        self.tmpdirname = tempfile.mkdtemp()
+
+        image_processor = CLIPImageProcessor()
+        tokenizer = LlamaTokenizerFast.from_pretrained("huggyllama/llama-7b")
+        processor_kwargs = self.prepare_processor_dict()
+        processor = LlavaProcessor(image_processor, tokenizer, **processor_kwargs)
+        processor.save_pretrained(self.tmpdirname)
+
+    def get_tokenizer(self, **kwargs):
+        return LlavaProcessor.from_pretrained(self.tmpdirname, **kwargs).tokenizer
+
+    def get_image_processor(self, **kwargs):
+        return LlavaProcessor.from_pretrained(self.tmpdirname, **kwargs).image_processor
+
+    def prepare_processor_dict(self):
+        return {"chat_template": "dummy_template"}
+
+    def test_processor_to_json_string(self):
+        processor = self.get_processor()
+        obj = json.loads(processor.to_json_string())
+        for key, value in self.prepare_processor_dict().items():
+            # chat templates are popped from dict
+            if key != "chat_template":
+                self.assertEqual(obj[key], value)
+                self.assertEqual(getattr(processor, key, None), value)
```

Comment:
Let's also assert the chat template is popped from the dict. 

```suggestion
        for key, value in self.prepare_processor_dict().items():
            # chat templates are popped from dict
            self.assertFalse(key == "chat_template")
            self.assertEqual(obj[key], value)
            self.assertEqual(getattr(processor, key, None), value)
```

Reply:
oops, no, the test fails because the `processor_dict` for llava has a `chat_template` key, and we use it in other tests for init and save the processor for ex. This test is same as the general one, with the exception that chat templates cannot pass `self.assertEqual(obj[key], value)` check

So we just want to test all other processor kwargs except chat template, which is tested separately. By other kwargs, I mean the ones which will be added soon

Reply:
> with the exception that chat templates cannot pass self.assertEqual(obj[key], value) check

I see, I missed what was happening originally in the test. Isn't `self.prepare_processor_dict().items()` a bit redundant, as we force `self.prepare_processor_dict()` to only have one key, which is `"chat_template"` and so all of this logic is skipped?

Reply:
Yes, same way as almost all processors skip this test. For VLMs this test will become available when we enforce new processing logic for input expansion with image tokens. Until then, we can override it to prevent failing tests, instead of `unitest.skip`

Reply:
hmmmm, I don't think overriding to make it look like tests are passing is a great idea. Skipping is far better as it's easier to spot and track. 

Part of the issue here is that this new behaviour still isn't being tested then, as we want to make sure that chat_template isn't in the processor_dict when saving out.

Reply:
Ah, we can add one more assert in  `test_chat_template_is_saved` to check what is the content of `processor_dict`

Oke, I'll skip it then with a comment explaining why and that we need to stop skipping it at some point

Reply:
Perfect :)